src/test/java/streams/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,127 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.jupiter.api.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+class Exercise1 {
+
+    @Test
+    void findPersonsEverWorkedInEpam() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toList()
+        List<Person> personsEverWorkedInEpam = employees.stream()
+                                                        .filter(employee -> employee.getJobHistory()
+                                                                                    .stream()
+                                                                                    .anyMatch(entry -> entry.getEmployer()
+                                                                                                            .equalsIgnoreCase("EPAM")))
+                                                        .map(Employee::getPerson)
+                                                        .collect(Collectors.toList());
+
+        assertThat(personsEverWorkedInEpam, contains(
+                employees.get(0).getPerson(),
+                employees.get(1).getPerson(),
+                employees.get(4).getPerson(),
+                employees.get(5).getPerson())
+        );
+    }
+
+    @Test
+    void findPersonsBeganCareerInEpam() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toList()
+        List<Person> startedFromEpam = employees.stream()
+                                                .filter(employee -> {
+                                                    Optional<JobHistoryEntry> firstJob = employee.getJobHistory().stream().findFirst();
+                                                    return firstJob.isPresent() && firstJob.get().getEmployer().equalsIgnoreCase("EPAM");
+                                                })
+                                                .map(Employee::getPerson)
+                                                .collect(Collectors.toList());
+        assertThat(startedFromEpam, contains(
+                employees.get(0).getPerson(),
+                employees.get(1).getPerson(),
+                employees.get(4).getPerson()
+        ));
+    }
+
+    @Test
+    void findAllCompanies() {
+        List<Employee> employees = getEmployees();
+
+        // TODO реализация, использовать Collectors.toSet()
+        Set<String> companies = employees.stream()
+                                         .flatMap(employee -> employee.getJobHistory().stream())
+                                         .map(JobHistoryEntry::getEmployer)
+                                         .distinct()
```

Comment:
`distinct()` is unnecessary operation here, because u're collecting elements into set.